### PR TITLE
This is an APICompat prototype

### PR DESCRIPTION
**Pull request by pakrym**
Referencing package directly doesn't work so I did this.

Calling dotnet build on `KeyVault.Secrets` runs APICompat against the package defined in packages.props.

### CODE REVIEW

File: eng/ApiCompat/ApiCompat.csproj [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<Project Sdk="Microsoft.NET.Sdk" DefaultTargets="VerifyCompat">
+
+    <PropertyGroup>
+        <TargetFramework>netstandard2.0</TargetFramework>
+        <RunApiCompatForSrc>true</RunApiCompatForSrc>
+    </PropertyGroup>
+
+    <ItemGroup>
+        <PackageReference Include="Microsoft.DotNet.ApiCompat" />
+        <PackageReference Include="Azure.Security.KeyVault.Secrets" />
```
Comment (pakrym) [inline eng/ApiCompat/ApiCompat.csproj]: We would have to reference all the packages we want to be able to validate here.

**Comment by JoshLove-msft** [comment]
@weshaggard what are your thoughts on this approach?

File: eng/Directory.Build.Data.targets [hunk 2]
```diff
@@ -116,5 +116,12 @@
            Text="When UseProjectReferenceToAzureClients=true all Azure.* references should be Project References, but the following are not [@(ShouldBeProjectReference)]" />
   </Target>
 
+  <Target Name="RunApiCompat" AfterTargets="CoreBuild">
```
Comment (JoshLove-msft) [inline eng/Directory.Build.Data.targets]: Add condition using ApiListing param

**Comment by weshaggard** [comment]
The general approach looks reasonable but we have to come up with a strategy to do the restore of the older packages.

**Pull request closed without merging**